src/main/scala/org/broadinstitute/dsde/workbench/sam/service/TosService.scala

```diff
@@ -60,9 +62,51 @@ class TosService(val directoryDao: DirectoryDAO, val tosConfig: TermsOfServiceCo
   private def isRollingWindowInEffect() = tosConfig.rollingAcceptanceWindowExpiration.exists(Instant.now().isBefore(_))
 
   @Deprecated
-  def getTosDetails(samUser: SamUser, samRequestContext: SamRequestContext): IO[TermsOfServiceDetails] =
+  def getTosDetails(samUser: SamUser, samRequestContext: SamRequestContext): IO[OldTermsOfServiceDetails] =
     directoryDao.getUserTos(samUser.id, samRequestContext).map { tos =>
-      TermsOfServiceDetails(isEnabled = true, tosConfig.isGracePeriodEnabled, tosConfig.version, tos.map(_.version))
+      OldTermsOfServiceDetails(isEnabled = true, tosConfig.isGracePeriodEnabled, tosConfig.version, tos.map(_.version))
+    }
+
+  def getTermsOfServiceDetailsForUser(
+      requestedUserId: WorkbenchUserId,
+      requestingUser: SamUser,
+      samRequestContext: SamRequestContext
+  ): IO[TermsOfServiceDetails] = {
+    def loadUserTosDetails =
+      for {
+        currentSamUserTos <- loadTosRecordForUser(requestedUserId, Option(tosConfig.version), samRequestContext)
+        previousSamUserTos <- loadTosRecordForUser(requestedUserId, tosConfig.previousVersion, samRequestContext)
+        requestedUser <- loadUser(requestedUserId, samRequestContext)
+      } yield TermsOfServiceDetails(
+        currentSamUserTos.version,
+        currentSamUserTos.createdAt,
+        tosAcceptancePermitsSystemUsage(requestedUser, Option(currentSamUserTos), Option(previousSamUserTos))
+      )
+
+    if (requestedUserId != requestingUser.id) {
```

Comment:
I have a feeling that this is going to be a common pattern. Could we pull it out into some sort of helper function. It would be nice to write something along the lines of
``` 
ensureNecessaryRights(samRequestContext) {
myLogic...
}
```